src/core/common/string.cpp

```diff
@@ -89,13 +89,16 @@ MatchType Match(const char *aString, const char *aPrefixString, StringMatchMode
 
 uint16_t StringLength(const char *aString, uint16_t aMaxLength)
 {
-    uint16_t ret;
+    uint16_t ret = 0;
 
-    for (ret = 0; (ret < aMaxLength) && (aString[ret] != kNullChar); ret++)
+    VerifyOrExit(aString != nullptr);
```

Comment:
The change LGTM. May be good to add this behavior to its documentaion in `string.hpp`

Also just wonderfing whether any existing code in OT  would call this with `nullptr`? How did you face this?

Reply:
Sure! Added.
I've started on implementing some unittests including corner cases for TCAT. I've managed to cause a memfault when setting only `mProvisioningUrl` in `aVendorInfo`, leaving `mPskdString`. It occured here:
https://github.com/openthread/openthread/blob/main/src/core/meshcop/tcat_agent.cpp#L86
https://github.com/openthread/openthread/blob/main/src/core/meshcop/meshcop.cpp#L90